.buildkite/scripts/cloud-cleanup-oidc.sh

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env bash
+
+source .buildkite/scripts/install_deps.sh
+
+cleanup_cloud_stale() {
+    local exit_code=$?
+
+    cd "$WORKSPACE"
+    rm -f "${AWS_RESOURCES_FILE}"
+    rm -f "${AWS_REDSHIFT_RESOURCES_FILE}"
+
+    exit "$exit_code"
+}
+
+trap cleanup_cloud_stale EXIT
+
+set -euo pipefail
+
+AWS_RESOURCES_FILE="aws.resources.txt"
+AWS_REDSHIFT_RESOURCES_FILE="redshift_clusters.json"
+
+RESOURCE_RETENTION_PERIOD="${RESOURCE_RETENTION_PERIOD:-"24 hours"}"
+DELETE_RESOURCES_BEFORE_DATE=$(date -Is -d "${RESOURCE_RETENTION_PERIOD} ago")
+export DELETE_RESOURCES_BEFORE_DATE
+
+CLOUD_REAPER_IMAGE="${DOCKER_REGISTRY}/observability-ci/cloud-reaper:0.3.0"
+
+DRY_RUN="$(buildkite-agent meta-data get DRY_RUN --default "${DRY_RUN:-"true"}")"
+
+resources_to_delete=0
+
+COMMAND="validate"
+if [[ "${DRY_RUN}" != "true" ]]; then
+    # TODO: to be changed to "destroy --confirm" once it can be tested
+    # that filters work as expected
+    COMMAND="plan"
+else
+    COMMAND="plan"
+fi
+
+any_resources_to_delete() {
+    local file=$1
+    local number=0
+    # First three lines are like:
+    # ⇒ Loading configuration...
+    # ✓ Succeeded to load configuration
+    # Scanning resources... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
+    number=$(tail -n +4 "${file}" | wc -l)
+    if [ "${number}" -eq 0 ]; then
+        return 1
+    fi
+    return 0
+}
+
+# As long as cloud reaper does not support OIDC authentication.
+create_aws_ephemeral_user() {
+    # Generate a unique name for the ephemeral IAM user.
+    EPHEMERAL_USER="ephemeral-admin-$(date +%s)"
+    echo "Creating IAM user: ${EPHEMERAL_USER}"
+    aws iam create-user --user-name "${EPHEMERAL_USER}" \
+        --tags Key=ephemeral,Value=true Key=division,Value=engineering Key=org,Value=obs Key=environment,Value=ci Key=repo,Value=elastic-package Key=created_at,Value="$(date -Is)"
+
+    echo "Attaching AdministratorAccess policy to ${EPHEMERAL_USER}..."
+    aws iam attach-user-policy --user-name "${EPHEMERAL_USER}" --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+
+    echo "Creating access keys for ${EPHEMERAL_USER}..."
+    creds_json=$(aws iam create-access-key --user-name "${EPHEMERAL_USER}")
+    AWS_ACCESS_KEY_ID_EPHEMERAL=$(echo "$creds_json" | jq -r '.AccessKey.AccessKeyId')
+    AWS_SECRET_ACCESS_KEY_EPHEMERAL=$(echo "$creds_json" | jq -r '.AccessKey.SecretAccessKey')
+    export EPHEMERAL_USER AWS_ACCESS_KEY_ID_EPHEMERAL AWS_SECRET_ACCESS_KEY_EPHEMERAL
```

Comment:
Should `AWS_ACCESS_KEY_ID_EPHEMERAL` and `AWS_SECRET_ACCESS_KEY_EPHEMERAL` be renamed to follow the patterns so their contents are redacted ? 

https://buildkite.com/docs/pipelines/configure/managing-log-output#redacted-environment-variables

Maybe something like `AWS_ACCESS_KEY_ID_EPHEMERAL_SECRET` and `AWS_SECRET_ACCESS_KEY_EPHEMERAL_SECRET` ? Or is it not needed in this scenario ?

Reply:
It does not matter at all, those env variables are not masked, unless they are set in the `pre-command`

Reply:
Ok, got it!